tools/arm_pack_manager/index.json

```diff
@@ -157016,6 +157016,110 @@
         ], 
         "vendor": "NXP:11"
     }, 
+    "LPC55S69JBD100": {
+        "algorithms": [
+            {
+                "default": false,
+                "file_name": "arm/LPC55XX_640.FLM",
+                "ram_size": 4096,
+                "ram_start": 536870912,
+                "size": 622592,
+                "start": 0
+            }
+        ],
+        "from_pack": {
+            "pack": "LPC55S69_DFP",
+            "url": "http://mcuxpresso.nxp.com/cmsis_pack/repo/",
+            "vendor": "NXP",
+            "version": "11.0.0"
+        },
+        "memories": {
+            "PROGRAM_FLASH": {
+                "access": {
+                    "execute": true,
+                    "non_secure": false,
+                    "non_secure_callable": false,
+                    "peripheral": false,
+                    "read": true,
+                    "secure": false,
+                    "write": false
+                },
+                "default": true,
+                "size": 622592,
+                "start": 0,
+                "startup": true
+            },
+            "SRAMX": {
+                "access": {
+                    "execute": false,
+                    "non_secure": false,
+                    "non_secure_callable": false,
+                    "peripheral": false,
+                    "read": true,
+                    "secure": false,
+                    "write": true
+                },
+                "default": false,
+                "size": 32768,
+                "start": 67108864,
+                "startup": false
+            },
+            "SRAM": {
+                "access": {
+                    "execute": false,
+                    "non_secure": false,
+                    "non_secure_callable": false,
+                    "peripheral": false,
+                    "read": true,
+                    "secure": false,
+                    "write": true
+                },
+                "default": true,
+                "size": 278528,
+                "start": 536870912,
+                "startup": false
+            },
+            "USB_RAM": {
+                "access": {
+                    "execute": false,
+                    "non_secure": false,
+                    "non_secure_callable": false,
+                    "peripheral": false,
+                    "read": true,
+                    "secure": false,
+                    "write": true
+                },
+                "default": false,
+                "size": 16384,
+                "start": 1074790400,
+                "startup": false
+            }
+        },
+        "name": "LPC55S69JBD100:cm33_core0",
```

Comment:
This was not generated by CPM. please use the definition from CPM

Reply:
If do not use this name then the device selection is not made. When I set the name to this value, then it make the right device selection.
![Device](https://user-images.githubusercontent.com/16884721/54287799-5967ab80-4574-11e9-921a-6d17099104ba.png)

Reply:
We can do this another way without manually editing the index. I'll have comments on how soon.

Reply:
I recommend:
```diff
diff --git a/tools/arm_pack_manager/index.json b/tools/arm_pack_manager/index.json
index 04099199fc..0e444f5400 100644
--- a/tools/arm_pack_manager/index.json
+++ b/tools/arm_pack_manager/index.json
@@ -157095,7 +157095,7 @@
                 "startup": false
             }
         },
-        "name": "LPC55S69JBD100:cm33_core0",
+        "name": "LPC55S69JBD100",
         "processor": {
             "Asymmetric": {
                 "cm33_core0": {
@@ -462779,4 +462779,4 @@
         "vendor": "Nordic Semiconductor:54"
     },
     "version": "0.2.0"
-}
\ No newline at end of file
+}
diff --git a/tools/export/cmsis/__init__.py b/tools/export/cmsis/__init__.py
index f4e4d94a18..0248c4f72e 100644
--- a/tools/export/cmsis/__init__.py
+++ b/tools/export/cmsis/__init__.py
@@ -51,10 +51,12 @@ class DeviceCMSIS():
         try:
             self.dfpu = target_info['processor']['Symmetric']['fpu']
         except KeyError:
-            cmsis_core = self.core.replace("F", "").replace("-", "")
-            for proc in target_info['processor']['Asymmetric'].values():
+            cmsis_core = self.core.replace("F", "").replace("-", "").replace("E", "")
+            for core_name, proc in target_info['processor']['Asymmetric'].items():
                 if proc['core'] == cmsis_core:
                     self.dfpu = proc['fpu']
+                    self.dname = '{}:{}'.format(self.dname, core_name)
+                    break
         self.debug, self.dvendor = self.vendor_debug(
             target_info.get('vendor') or target_info['from_pack']['vendor']
         )
```

I verified that this procudes the same `LPC55S69JBD100:cm33_core0` string in the project file.

Reply:
Great thank you, I will update the name. Please help check the other issue with memory section.

Reply:
This change does not seem to be working
-            self.dfpu = target_info['processor']['Asymmetric']['fpu']
+            cmsis_core = self.core.replace("F", "").replace("-", "").replace("E", "")
+            for core_name, proc in target_info['processor']['Asymmetric'].items():
+                if proc['core'] == cmsis_core:
+                    self.dfpu = proc['fpu']
+                    self.dname = '{}:{}'.format(self.dname, core_name)
+                    break
+                print(self.dname)

PS C:\mbed-os-example-blinky> mbed export -i uvision6 -m LPC55S69_NS
[mbed] Working path "C:\mbed-os-example-blinky" (program)
Scan: mbed-os-example-blinky
LPC55S69JBD100
LPC55S69JBD100

Reply:
:+1: Good catch, I was uisng the `_S` variant.

```diff
diff --git a/tools/arm_pack_manager/index.json b/tools/arm_pack_manager/index.json
index 04099199fc..0e444f5400 100644
--- a/tools/arm_pack_manager/index.json
+++ b/tools/arm_pack_manager/index.json
@@ -157095,7 +157095,7 @@
                 "startup": false
             }
         },
-        "name": "LPC55S69JBD100:cm33_core0",
+        "name": "LPC55S69JBD100",
         "processor": {
             "Asymmetric": {
                 "cm33_core0": {
@@ -462779,4 +462779,4 @@
         "vendor": "Nordic Semiconductor:54"
     },
     "version": "0.2.0"
-}
\ No newline at end of file
+}
diff --git a/tools/export/cmsis/__init__.py b/tools/export/cmsis/__init__.py
index f4e4d94a18..a4569122b5 100644
--- a/tools/export/cmsis/__init__.py
+++ b/tools/export/cmsis/__init__.py
@@ -51,10 +51,14 @@ class DeviceCMSIS():
         try:
             self.dfpu = target_info['processor']['Symmetric']['fpu']
         except KeyError:
-            cmsis_core = self.core.replace("F", "").replace("-", "")
-            for proc in target_info['processor']['Asymmetric'].values():
+            # TODO: refactor this into a "base_core_for" function
+            cmsis_core = self.core.replace("F", "").replace("-", "").replace("E", "")
+            cmsis_core = cmsis_core.replace("NS", "")
+            for core_name, proc in target_info['processor']['Asymmetric'].items():
                 if proc['core'] == cmsis_core:
                     self.dfpu = proc['fpu']
+                    self.dname = '{}:{}'.format(self.dname, core_name)
+                    break
         self.debug, self.dvendor = self.vendor_debug(
             target_info.get('vendor') or target_info['from_pack']['vendor']
         )
```

Reply:
Thanks this works. The Device Tab of the Project Options is set correctly. The Target Tab of the Properties is not correct.

Reply:
It's a start. I'll look into what it would take to set the Target Tab correctly.